test/ExportQueryExecutionTreesTest.cpp

```diff
@@ -1637,3 +1637,94 @@ TEST(ExportQueryExecutionTrees, convertGeneratorForChunkedTransfer) {
               AllOf(HasSubstr("!!!!>># An error has occurred"),
                     HasSubstr("A very strange")));
 }
+
+TEST(ExportQueryExecutionTrees, idToLiteralOrIriFunctionality) {
+  std::string kg =
+      "<s> <p> \"something\" . <s> <p> 1. <s> <p> "
+      "\"some^^<http://www.w3.org/2001/XMLSchema#string>\" .";
+  auto qec = ad_utility::testing::getQec(kg);
+  auto getId = ad_utility::testing::makeGetId(qec->getIndex());
+  using enum Datatype;
+
+  // Case Literal With no Datatype
+  {
+    Id id = getId("\"something\"");
+    auto resultLiteral = ExportQueryExecutionTrees::idToLiteralOrIri(
+        qec->getIndex(), id, LocalVocab{});
+    EXPECT_EQ(resultLiteral.value().toStringRepresentation(), "\"something\"");
+    // Case onlyReturnLiterals
+    resultLiteral = ExportQueryExecutionTrees::idToLiteralOrIri<true>(
+        qec->getIndex(), id, LocalVocab{});
+    EXPECT_EQ(resultLiteral.value().toStringRepresentation(), "\"something\"");
+    // Case onlyReturnLiteralsWithXsdString
+    resultLiteral = ExportQueryExecutionTrees::idToLiteralOrIri(
+        qec->getIndex(), id, LocalVocab{}, true);
+    EXPECT_EQ(resultLiteral.value().toStringRepresentation(), "\"something\"");
+  }
+
+  // Case Literal With Datatype String
+  {
+    Id id = getId("\"some^^<http://www.w3.org/2001/XMLSchema#string>\"");
+    auto resultLiteral = ExportQueryExecutionTrees::idToLiteralOrIri(
+        qec->getIndex(), id, LocalVocab{});
+    EXPECT_EQ(resultLiteral.value().toStringRepresentation(),
+              "\"some^^<http://www.w3.org/2001/XMLSchema#string>\"");
+    // TODO: Problem: The Literal has no Datatype
+    EXPECT_EQ(resultLiteral.value().hasDatatype(), false);
+    // Case onlyReturnLiterals
+    resultLiteral = ExportQueryExecutionTrees::idToLiteralOrIri<true>(
+        qec->getIndex(), id, LocalVocab{});
+    EXPECT_EQ(resultLiteral.value().toStringRepresentation(),
+              "\"some^^<http://www.w3.org/2001/XMLSchema#string>\"");
+    // Case onlyReturnLiteralsWithXsdString
+    resultLiteral = ExportQueryExecutionTrees::idToLiteralOrIri(
+        qec->getIndex(), id, LocalVocab{}, true);
+    EXPECT_EQ(resultLiteral.value().toStringRepresentation(),
+              "\"some^^<http://www.w3.org/2001/XMLSchema#string>\"");
+  }
+
+  // TODO: Case Literal With Datatype not equal String
```

Comment:
Exactly the same thing, use `"\"some\"^^<http://www.dadudeldu.com/NoSuchDatatype>"`, It was only a typo in the syntax of your literals with the escaping..